sky/provision/docker_utils.py

```diff
@@ -340,9 +343,12 @@ def _docker_expand_user(self, string, any_char=False):
         user_pos = string.find('~')
         if user_pos > -1:
             if self.home_dir is None:
-                self.home_dir = (self._run(
+                self.home_dir = self._run(
                     f'{self.docker_cmd} exec {self.container_name} '
-                    'printenv HOME',))
+                    'printenv HOME',
+                    separate_stderr=True)
+                assert '\n' not in self.home_dir, (
+                    f'Unexpected newline in HOME env variable: {self.home_dir}')
```

Comment:
Should we raising an error instead?

Reply:
We may want to keep it as assert since it should not happen with the latest fix, and assertion is used for catching unexpected behavior

Reply:
Sounds good!